src/app.ts

```diff
@@ -13,14 +14,60 @@ interface TemperatureSummary {
   average: number
 }
 
+const temperatures: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  temperatures.push(...readings)
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const temperatureSummary: TemperatureSummary = {
+    first: 0,
+    last: 0,
+    high: 0,
+    low: 0,
+    average: 0,
+  }
+
+  const filteredTemperatures = temperatures.filter(
+    (temp) =>
+      temp.time.getTime() === new Date(date).getTime() && temp.city === city,
+  )
+  const filteredTempLegnth: number = filteredTemperatures.length
+
+  if (filteredTempLegnth > 0) {
+    temperatureSummary.first = filteredTemperatures[0].temperature
+    temperatureSummary.last =
+      filteredTemperatures[filteredTempLegnth - 1].temperature
+  }
```

Comment:
by failing fast as I mentioned above, you can put this code outside of the `if` bloick